Strict read only mode

hg-ms opened the pull request:
A strict read only storage mode that can be enabled/disabled and does not allow any write actions

hg-ms committed:
first implementation

fh-ms commented on storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java:
```diff
@@ -0,0 +1,149 @@
+	// methods //
+	////////////
+	
+	public void enableReadOnly(final boolean enableReadOnly)
```
Probably better to use getter/setter style naming, here setReadOnly(boolean)

hg-ms replied:
ok, done

fh-ms commented on storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java:
```diff
@@ -0,0 +1,149 @@
+		logger.info("Read-only mode enabled: {}", this.writeController == this.readOnlyController);
+	}
+	
+	public boolean readOnlyEnabled()
```
Probably better to use getter/setter style naming, here isReadOnly()

hg-ms replied:
ok, done

fh-ms commented on storage/storage/src/main/java/one/microstream/storage/types/StorageWriteControllerReadOnlyMode.java:
```diff
@@ -0,0 +1,149 @@
+		this.readOnlyController = WriteController.Disabled();
+		this.writeController = this.readOnlyController;
+		
+		logger.info("Created SwitchableStorageWriteController with Storage read-only mode enabled: {}!",
```
StorageWriteControllerReadOnlyMode != SwitchableStorageWriteController

hg-ms replied:
ok, done

hg-ms committed:
fixed review findings

renamed enableReadOnly to setReadOnly
renamed readOnlyEnabled to isReadOnly
fixed logging text
